$NetBSD: patch-af,v 1.6 2007/02/23 03:37:55 rillig Exp $

The CPPFLAGS and LIBS from wx-config had been set too early in the
configure file, so that they were taken into the AC_CONFIG_SUBDIRS
directories. Some of these directories only use a C compiler for
linking, which of course caused problems with the wx* libraries, which
are C++.

--- configure.orig	2006-11-15 05:12:13.000000000 +0100
+++ configure	2007-02-23 04:17:41.000000000 +0100
@@ -6586,7 +6586,8 @@ fi
 subdirs="$subdirs lib-src/portaudio-v19"
 
 
-else
+fi
+if [ "$use_portaudio" = "v18" ] ; then
 
 	CXXFLAGS="-I../lib-src/portaudio/pa_common $CXXFLAGS"
 
@@ -6672,6 +6673,84 @@ _ACEOF
 
 	esac
 fi
+if [ "$use_portaudio" = "system" ] ; then
+
+	LIBS="$LIBS -lportaudio"
+
+	case "${host_os}" in
+	  darwin* | rhapsody*)
+
+		CONFIGHEADER="configunix.h"
+
+
+		CXXFLAGS="-I../mac $CXXFLAGS"
+
+	   if [ "$use_portmixer" = "yes" ] ; then
+			BUILDLIBS="$BUILDLIBS portmixer/px_mac_core/portmixer.a"
+			EXTRAOBJS="$EXTRAOBJS ../lib-src/portmixer/px_mac_core/portmixer.a"
+	   fi
+
+		LIBS="-framework AudioUnit -framework AudioToolbox $LIBS"
+		LIBS="-framework CoreAudio $LIBS -lz"
+
+		EXTRATARGETS="../Audacity.app"
+
+		if [ "$use_ladspa" = "default" ] ; then
+			use_ladspa="yes";
+		fi
+
+
+		OPTOBJS="$OPTOBJS \$(OBJDIR)/effects/VST/LoadVSTMac.o"
+		OPTOBJS="$OPTOBJS \$(OBJDIR)/effects/VST/VSTEffect.o"
+
+	;;
+
+	cygwin*)
+
+ 	LIBS="$LIBS -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 -lwsock32 -lwinmm"
+	LIBS="$LIBS -Wl,--subsystem,windows"
+ 	CONFIGHEADER="configwin.h"
+
+    if [ "$use_portmixer" = "yes" ] ; then
+ 		BUILDLIBS="$BUILDLIBS portmixer/px_win_wmme/portmixer.a"
+ 		EXTRAOBJS="$EXTRAOBJS ../lib-src/portmixer/px_win_wmme/portmixer.a Audacity.coff"
+    fi
+
+ 	if [ "$use_ladspa" = "default" ] ; then
+ 		use_ladspa="yes";
+ 	fi
+
+
+cat >>confdefs.h <<\_ACEOF
+#define __CYGWIN__ 1
+_ACEOF
+
+
+cat >>confdefs.h <<\_ACEOF
+#define _FILE_OFFSET_BITS 32
+_ACEOF
+
+
+ 	AFTERBUILD=""
+	;;
+
+	  *)
+
+		CONFIGHEADER="configunix.h"
+
+	   if [ "$use_portmixer" = "yes" ] ; then
+			BUILDLIBS="$BUILDLIBS portmixer/px_unix_oss/portmixer.a"
+			EXTRAOBJS="$EXTRAOBJS ../lib-src/portmixer/px_unix_oss/portmixer.a"
+	   fi
+
+		if [ "$use_ladspa" = "default" ] ; then
+			use_ladspa="yes";
+		fi
+
+		AFTERBUILD=""
+
+	esac
+fi
 
 use_precomp="no"
 
